superset/viz.py

```diff
@@ -1199,10 +1199,10 @@ def process_data(self, df, aggregate=False):
 
     def run_extra_queries(self):
         fd = self.form_data
-        time_compare = fd.get('time_compare')
-        if time_compare:
+        time_compare = fd.get('time_compare') or []
+        for option in time_compare:
```

Comment:
Eventually (out of scope for this PR) we could have parallelization mechanism for this kind of things, making timeouts less likely.

Reply:
(y)

Reply:
I looked into parallelizing this, but it's non trivial: we can't just wrap the calls here in threads or processes because the connection is already established at this point, and for some backends like SQLite it can't be shared between threads/processes.

I agree that we should probably do this in a separate PR, making the logic of [`run_extra_queries`](https://github.com/apache/incubator-superset/blob/fa3e4e23b331dad2682a8a005facb0a55229cb72/superset/viz.py#L125-L146) more robust.